pennylane/interfaces/set_shots.py

```diff
@@ -40,6 +43,11 @@ def set_shots(device, shots):
     >>> set_shots(dev, shots=100)(lambda: dev.shots)()
     100
     """
+    if isinstance(device, qml.devices.experimental.Device):

```

Comment:
any reason to not raise a warning here? I feel like a user might do something like:
```python
with set_shots(dq2, 100):
    @qml.qnode(dq2)
    def circuit():
        ...
```
EDIT: I see now that it's used in qnode's `call`, so I don't wanna just raise a warning. but still figured it's worth pointing out

Reply:
It's a fairly internal function, so I'm not too concerned about a user accessing it.

And it cleans up a lot of code where we would otherwise need to do a bunch of `if-else` branching.